Library/Source/NativeEngine.cpp

```diff
@@ -1188,14 +1254,44 @@ namespace babylon
 
     Napi::Value NativeEngine::GetRenderWidth(const Napi::CallbackInfo& info)
     {
-        // TODO CHECK: Is this not just the size?  What is this?
-        return Napi::Value::From(info.Env(), m_renderTargetSize.Width);
+        return Napi::Value::From(info.Env(), bgfx::getStats()->width);
     }
 
     Napi::Value NativeEngine::GetRenderHeight(const Napi::CallbackInfo& info)
     {
-        // TODO CHECK: Is this not just the size?  What is this?
-        return Napi::Value::From(info.Env(), m_renderTargetSize.Height);
+        return Napi::Value::From(info.Env(), bgfx::getStats()->height);
+    }
+
+    void NativeEngine::SetViewPort(const Napi::CallbackInfo& info)
+    {
+        const auto x = info[0].As<Napi::Number>().FloatValue();
+        const auto y = info[1].As<Napi::Number>().FloatValue();
+        const auto width = info[2].As<Napi::Number>().FloatValue();
+        const auto height = info[3].As<Napi::Number>().FloatValue();
+
+        const auto backbufferWidth = bgfx::getStats()->width;
+        const auto backbufferHeight = bgfx::getStats()->height;
+        auto newViewId = m_viewidSet.Get();
```

Comment:
I don't understand this. Why do we need to keep ids around for view ports? Can we have more than one?

Reply:
Also, it seems like the recycle set is being passed to the FrameBufferManager also. Not sure this is right? It will affect CreateFrameBuffer and XR stuff.

Reply:
a bgfx view only store 1 viewRect/1 framebuffer. To use more viewRects, we use more views. Multiple view can access the same frameBuffer. To not mixup view ids during a frame, we have to mutualize the ids with the FrameBufferManager.

Reply:
Sounds like the assumptions I made when I wrote the `FrameBufferManager` -- namely, a one-to-one relationship between views and framebuffers -- were incorrect. The `FrameBufferManager` mostly existed to hide the `RecycleSet` as an implementation detail; so if we're having to expose the `RecycleSet` to other things, that might be a sign we need need to reconsider how that system is architected. That might be a task for a different PR, though.